In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df=pd.read_csv('https://raw.githubusercontent.com/niteen11/DataAnalyticsAcademy/master/Python/dataset_diabetes/diabetic_data.csv'
)

In [3]:
df

encounter_id  patient_nbr             race  gender      age weight  \
0            2278392      8222157        Caucasian  Female   [0-10)      ?   
1             149190     55629189        Caucasian  Female  [10-20)      ?   
2              64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3             500364     82442376        Caucasian    Male  [30-40)      ?   
4              16680     42519267        Caucasian    Male  [40-50)      ?   
...              ...          ...              ...     ...      ...    ...   
101761     443847548    100162476  AfricanAmerican    Male  [70-80)      ?   
101762     443847782     74694222  AfricanAmerican  Female  [80-90)      ?   
101763     443854148     41088789        Caucasian    Male  [70-80)      ?   
101764     443857166     31693671        Caucasian  Female  [80-90)      ?   
101765     443867222    175429310        Caucasian    Male  [70-80)      ?   

        admission_type_id  discharge_disposition_id  admission_source_id  \
0                       6                        25                    1   
1                       1                         1                    7   
2                       1                         1                    7   
3                       1                         1                    7   
4                       1                         1                    7   
...                   ...                       ...                  ...   
101761                  1                         3                    7   
101762                  1                         4                    5   
101763                  1                         1                    7   
101764                  2                         3                    7   
101765                  1                         1                    7   

        time_in_hospital  ... citoglipton insulin  glyburide-metformin  \
0                      1  ...          No      No                   No   
1                      3  ...          No      Up                   No   
2                      2  ...          No      No                   No   
3                      2  ...          No      Up                   No   
4                      1  ...          No  Steady                   No   
...                  ...  ...         ...     ...                  ...   
101761                 3  ...          No    Down                   No   
101762                 5  ...          No  Steady                   No   
101763                 1  ...          No    Down                   No   
101764                10  ...          No      Up                   No   
101765                 6  ...          No      No                   No   

        glipizide-metformin  glimepiride-pioglitazone  \
0                        No                        No   
1                        No                        No   
2                        No                        No   
3                        No                        No   
4                        No                        No   
...                     ...                       ...   
101761                   No                        No   
101762                   No                        No   
101763                   No                        No   
101764                   No                        No   
101765                   No                        No   

        metformin-rosiglitazone  metformin-pioglitazone  change diabetesMed  \
0                            No                      No      No          No   
1                            No                      No      Ch         Yes   
2                            No                      No      No         Yes   
3                            No                      No      Ch         Yes   
4                            No                      No      Ch         Yes   
...                         ...                     ...     ...         ...   
101761                       No

# VARIABLE NAMES: DESCRIPTION 
●	Encounter ID: Unique identifier of an encounter  
●	Patient number: Unique identifier of a patient  
●	Race Values: Caucasian, Asian, African American, Hispanic, and other  
●	Gender Values: male, female, and unknown/invalid  
●	Age Grouped in 10-year intervals: 0, 10), 10, 20), …, 90, 100)  
●	Weight: Weight in pounds  
●	Admission type: Integer identifier corresponding to 9 distinct values, for example, emergency, urgent, elective, newborn, and not available  
●	Discharge disposition: Integer identifier corresponding to 29 distinct values, for example, discharged to home, expired, and not available  
●	Admission source: Integer identifier corresponding to 21 distinct values, for example, physician referral, emergency room, and transfer from a hospital  
●	Time in hospital: Integer number of days between admission and discharge  
●	Payer code: Integer identifier corresponding to 23 distinct values, for example, Blue Cross/Blue Shield, Medicare, and self-pay Medical  
●	Medical specialty: Integer identifier of a specialty of the admitting physician, corresponding to 84 distinct values, for example, cardiology, internal medicine, family/general practice, and surgeon  
●	Number of lab procedures: Number of lab tests performed during the encounter  
●	Number of procedures: Numeric Number of procedures (other than lab tests) performed during the encounter  
●	Number of medications: Number of distinct generic names administered during the encounter  
●	Number of outpatient visits: Number of outpatient visits of the patient in the year preceding the encounter  
●	Number of emergency visits: Number of emergency visits of the patient in the year preceding the encounter  
●	Number of inpatient visits: Number of inpatient visits of the patient in the year preceding the encounter  
●	Diagnosis 1: The primary diagnosis (coded as first three digits of ICD9); 848 distinct values  
●	Diagnosis 2: Secondary diagnosis (coded as first three digits of ICD9); 923 distinct values  
●	Diagnosis 3: Additional secondary diagnosis (coded as first three digits of ICD9); 954 distinct values  
●	Number of diagnoses Number of diagnoses entered to the system 0%  
●	Glucose serum test result Indicates the range of the result or if the test was not taken. Values: “>200,” “>300,” “normal,” and “none” if not measured  
●	A1c test result Indicates the range of the result or if the test was not taken. Values: “>8” if the result was greater than 8%, “>7” if the result was greater than 7% but less than 8%, “normal” if the result was less than 7%, and “none” if not measured.  
●	Change of medications Indicates if there was a change in diabetic medications (either dosage or generic name). Values: “change” and “no change”  
●	Diabetes medications Indicates if there was any diabetic medication prescribed. Values: “yes” and “no”  
●	24 features for medications For the generic names: metformin, repaglinide, nateglinide, chlorpropamide, glimepiride, acetohexamide, glipizide, glyburide, tolbutamide, pioglitazone, rosiglitazone, acarbose, miglitol, troglitazone, tolazamide, examide, sitagliptin, insulin, glyburide-metformin, glipizide-metformin, glimepiride- pioglitazone, metformin-rosiglitazone, and metformin- pioglitazone, the feature indicates whether the drug was prescribed or there was a change in the dosage. Values: “up” if the dosage was increased during the encounter, “down” if the dosage was decreased, “steady” if the dosage did not change, and “no” if the drug was not prescribed  
●	Readmitted Days to inpatient readmission. Values: “<30” if the patient was readmitted in less than 30 days, “>30” if the patient was readmitted in more than 30 days, and “No” for no record of readmission


In [4]:
#show the first 5 rows of the dataset
df.head()

encounter_id  patient_nbr             race  gender      age weight  \
0       2278392      8222157        Caucasian  Female   [0-10)      ?   
1        149190     55629189        Caucasian  Female  [10-20)      ?   
2         64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3        500364     82442376        Caucasian    Male  [30-40)      ?   
4         16680     42519267        Caucasian    Male  [40-50)      ?   

   admission_type_id  discharge_disposition_id  admission_source_id  \
0                  6                        25                    1   
1                  1                         1                    7   
2                  1                         1                    7   
3                  1                         1                    7   
4                  1                         1                    7   

   time_in_hospital  ... citoglipton insulin  glyburide-metformin  \
0                 1  ...          No      No                   No   
1                 3  ...          No      Up                   No   
2                 2  ...          No      No                   No   
3                 2  ...          No      Up                   No   
4                 1  ...          No  Steady                   No   

   glipizide-metformin  glimepiride-pioglitazone  metformin-rosiglitazone  \
0                   No                        No                       No   
1                   No                        No                       No   
2                   No                        No                       No   
3                   No                        No                       No   
4                   No                        No                       No   

   metformin-pioglitazone  change diabetesMed readmitted  
0                      No      No          No         NO  
1                      No      Ch         Yes        >30  
2                      No      No         Yes         NO  
3                      No      Ch         Yes         NO  
4                      No      Ch         Yes         NO  

[5 rows x 50 columns]

In [5]:
#the data set has 50 columns and 101766 rows
df.shape

(101766, 50)

In [6]:
df.columns

Index(['encounter_id', 'patient_nbr', 'race', 'gender', 'age', 'weight',
       'admission_type_id', 'discharge_disposition_id', 'admission_source_id',
       'time_in_hospital', 'payer_code', 'medical_specialty',
       'num_lab_procedures', 'num_procedures', 'num_medications',
       'number_outpatient', 'number_emergency', 'number_inpatient', 'diag_1',
       'diag_2', 'diag_3', 'number_diagnoses', 'max_glu_serum', 'A1Cresult',
       'metformin', 'repaglinide', 'nateglinide', 'chlorpropamide',
       'glimepiride', 'acetohexamide', 'glipizide', 'glyburide', 'tolbutamide',
       'pioglitazone', 'rosiglitazone', 'acarbose', 'miglitol', 'troglitazone',
       'tolazamide', 'examide', 'citoglipton', 'insulin',
       'glyburide-metformin', 'glipizide-metformin',
       'glimepiride-pioglitazone', 'metformin-rosiglitazone',
       'metformin-pioglitazone', 'change', 'diabetesMed', 'readmitted'],
      dtype='object')

In [7]:
df.describe

<bound method NDFrame.describe of         encounter_id  patient_nbr             race  gender      age weight  \
0            2278392      8222157        Caucasian  Female   [0-10)      ?   
1             149190     55629189        Caucasian  Female  [10-20)      ?   
2              64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3             500364     82442376        Caucasian    Male  [30-40)      ?   
4              16680     42519267        Caucasian    Male  [40-50)      ?   
...              ...          ...              ...     ...      ...    ...   
101761     443847548    100162476  AfricanAmerican    Male  [70-80)      ?   
101762     443847782     74694222  AfricanAmerican  Female  [80-90)      ?   
101763     443854148     41088789        Caucasian    Male  [70-80)      ?   
101764     443857166     31693671        Caucasian  Female  [80-90)      ?   
101765     443867222    175429310        Caucasian    Male  [70-80)      ?   

        admission_type_id  di

In [11]:
#select multiple columns of the dataset
df[['gender','age','medical_specialty','diag_1','diabetesMed', 'readmitted']]

gender      age         medical_specialty  diag_1 diabetesMed  \
0       Female   [0-10)  Pediatrics-Endocrinology  250.83          No   
1       Female  [10-20)                         ?     276         Yes   
2       Female  [20-30)                         ?     648         Yes   
3         Male  [30-40)                         ?       8         Yes   
4         Male  [40-50)                         ?     197         Yes   
...        ...      ...                       ...     ...         ...   
101761    Male  [70-80)                         ?  250.13         Yes   
101762  Female  [80-90)                         ?     560         Yes   
101763    Male  [70-80)                         ?      38         Yes   
101764  Female  [80-90)           Surgery-General     996         Yes   
101765    Male  [70-80)                         ?     530          No   

       readmitted  
0              NO  
1             >30  
2              NO  
3              NO  
4              NO  
...           ...  
101761        >30  
101762         NO  
101763         NO  
101764         NO  
101765         NO  

[101766 rows x 6 columns]

In [4]:
df['age'].count

<bound method Series.count of 0          [0-10)
1         [10-20)
2         [20-30)
3         [30-40)
4         [40-50)
           ...   
101761    [70-80)
101762    [80-90)
101763    [70-80)
101764    [80-90)
101765    [70-80)
Name: age, Length: 101766, dtype: object>

In [12]:
#sort the dataset according to the diag_1
df.sort_values('diag_1')

encounter_id  patient_nbr             race  gender       age  \
99611     415513862    152018339            Other    Male   [60-70)   
55425     160287738     52189857  AfricanAmerican    Male   [70-80)   
66525     185798838     60145218            Asian    Male   [50-60)   
81899     254546094    111103389        Caucasian    Male   [80-90)   
86984     277122066     69670206         Hispanic    Male   [50-60)   
...             ...          ...              ...     ...       ...   
95163     357921158    123973484        Caucasian  Female   [70-80)   
18414      66683040     23487408        Caucasian  Female   [70-80)   
63030     175382550     69256188        Caucasian    Male   [80-90)   
16366      61551600     25825491        Caucasian  Female  [90-100)   
38378     119314440     23255469        Caucasian    Male   [70-80)   

          weight  admission_type_id  discharge_disposition_id  \
99611          ?                  2                         1   
55425          ?                  1                         1   
66525          ?                  1                         1   
81899          ?                  2                         3   
86984          ?                  3                         1   
...          ...                ...                       ...   
95163          ?                  2                         3   
18414          ?                  5                         3   
63030  [100-125)                  2                         1   
16366          ?                  5                         3   
38378          ?                  1                         1   

       admission_source_id  time_in_hospital  ... citoglipton insulin  \
99611                    1                12  ...          No    Down   
55425                    7                13  ...          No      No   
66525                    7                 3  ...          No    Down   
81899                    7                 6  ...          No      No   
86984                    7                 4  ...          No  Steady   
...                    ...               ...  ...         ...     ...   
95163                    4                 2  ...          No  Steady   
18414                   17                 4  ...          No    Down   
63030                    1                 5  ...          No  Steady   
16366                   17                 3  ...          No      No   
38378                    7                 1  ...          No      No   

       glyburide-metformin  glipizide-metformin  glimepiride-pioglitazone  \
99611                   No                   No                        No   
55425                   No                   No                        No   
66525                   No                   No                        No   
81899                   No                   No                        No   
86984                   No                   No                        No   
...                    ...                  ...                       ...   
95163                   No                   No                        No   
18414                   No                   No                        No   
63030                   No                   No                        No   
16366                   No                   No                        No   
38378                   No                   No                        No   

       metformin-rosiglitazone  metformin-pioglitazone  change diabetesMed  \
99611                       No                      No      Ch         Yes   
55425                       No                      No      No          No   
66525                       No                      No      Ch         Yes   
81899                       No                      No      No          No   
86984                       No                      No      Ch         Yes   
...                        ...                     ...     ...         ...   
95163                    

In [15]:
#filter by age group
df[df['age']=='[60-70)']

encounter_id  patient_nbr             race  gender      age  \
6              55842     84259809        Caucasian    Male  [60-70)   
11             36900     77391171  AfricanAmerican    Male  [60-70)   
14             62256     49726791  AfricanAmerican  Female  [60-70)   
15             73578     86328819  AfricanAmerican    Male  [60-70)   
21            150048     21239181                ?    Male  [60-70)   
...              ...          ...              ...     ...      ...   
101723     443695760     98476776        Caucasian    Male  [60-70)   
101724     443698988    105807996  AfricanAmerican    Male  [60-70)   
101736     443739152     90751788        Caucasian  Female  [60-70)   
101756     443842070    140199494            Other  Female  [60-70)   
101760     443847176     50375628  AfricanAmerican  Female  [60-70)   

           weight  admission_type_id  discharge_disposition_id  \
6               ?                  3                         1   
11              ?                  2                         1   
14              ?                  3                         1   
15              ?                  1                         3   
21              ?                  2                         1   
...           ...                ...                       ...   
101723  [150-175)                  3                         6   
101724          ?                  6                         6   
101736          ?                  1                         3   
101756          ?                  1                         1   
101760          ?                  1                         1   

        admission_source_id  time_in_hospital  ... citoglipton insulin  \
6                         2                 4  ...          No  Steady   
11                        4                 7  ...          No  Steady   
14                        2                 1  ...          No  Steady   
15                        7                12  ...          No      Up   
21                        4                 2  ...          No  Steady   
...                     ...               ...  ...         ...     ...   
101723                    1                 3  ...          No      No   
101724                   17                 7  ...          No      No   
101736                    7                 8  ...          No  Steady   
101756                    7                 2  ...          No  Steady   
101760                    7                 6  ...          No    Down   

        glyburide-metformin  glipizide-metformin  glimepiride-pioglitazone  \
6                        No                   No                        No   
11                       No                   No                        No   
14                       No                   No                        No   
15                       No                   No                        No   
21                       No                   No                        No   
...                     ...                  ...                       ...   
101723                   No                   No                        No   
101724                   No                   No                        No   
101736                   No                   No                        No   
101756                   No                   No                        No   
101760                   No                   No                        No   

        metformin-rosiglitazone  metformin-pioglitazone  change diabetesMed  \
6                            No                      No      Ch         Yes   
11                           No                      No      Ch         Yes   
14                           No                      No      No         Yes   
15                           No                      No      Ch         Yes   
21                           No                      No      Ch         Yes   
...                         ...                     ...     ..

In [6]:
#filter by age group 60-70
df[df['age']=='[60-70)'].count()

encounter_id                22483
patient_nbr                 22483
race                        22483
gender                      22483
age                         22483
weight                      22483
admission_type_id           22483
discharge_disposition_id    22483
admission_source_id         22483
time_in_hospital            22483
payer_code                  22483
medical_specialty           22483
num_lab_procedures          22483
num_procedures              22483
num_medications             22483
number_outpatient           22483
number_emergency            22483
number_inpatient            22483
diag_1                      22483
diag_2                      22483
diag_3                      22483
number_diagnoses            22483
max_glu_serum               22483
A1Cresult                   22483
metformin                   22483
repaglinide                 22483
nateglinide                 22483
chlorpropamide              22483
glimepiride                 22483
acetohexamide 

In [7]:
#filter by age group 50-60
df[df['age']=='[50-60)'].count()

encounter_id                17256
patient_nbr                 17256
race                        17256
gender                      17256
age                         17256
weight                      17256
admission_type_id           17256
discharge_disposition_id    17256
admission_source_id         17256
time_in_hospital            17256
payer_code                  17256
medical_specialty           17256
num_lab_procedures          17256
num_procedures              17256
num_medications             17256
number_outpatient           17256
number_emergency            17256
number_inpatient            17256
diag_1                      17256
diag_2                      17256
diag_3                      17256
number_diagnoses            17256
max_glu_serum               17256
A1Cresult                   17256
metformin                   17256
repaglinide                 17256
nateglinide                 17256
chlorpropamide              17256
glimepiride                 17256
acetohexamide 

In [8]:
#filter by age group 40-50
df[df['age']=='[40-50)'].count()

encounter_id                9685
patient_nbr                 9685
race                        9685
gender                      9685
age                         9685
weight                      9685
admission_type_id           9685
discharge_disposition_id    9685
admission_source_id         9685
time_in_hospital            9685
payer_code                  9685
medical_specialty           9685
num_lab_procedures          9685
num_procedures              9685
num_medications             9685
number_outpatient           9685
number_emergency            9685
number_inpatient            9685
diag_1                      9685
diag_2                      9685
diag_3                      9685
number_diagnoses            9685
max_glu_serum               9685
A1Cresult                   9685
metformin                   9685
repaglinide                 9685
nateglinide                 9685
chlorpropamide              9685
glimepiride                 9685
acetohexamide               9685
glipizide 

In [9]:
#filter by age group 30-40
df[df['age']=='[30-40)'].count()

encounter_id                3775
patient_nbr                 3775
race                        3775
gender                      3775
age                         3775
weight                      3775
admission_type_id           3775
discharge_disposition_id    3775
admission_source_id         3775
time_in_hospital            3775
payer_code                  3775
medical_specialty           3775
num_lab_procedures          3775
num_procedures              3775
num_medications             3775
number_outpatient           3775
number_emergency            3775
number_inpatient            3775
diag_1                      3775
diag_2                      3775
diag_3                      3775
number_diagnoses            3775
max_glu_serum               3775
A1Cresult                   3775
metformin                   3775
repaglinide                 3775
nateglinide                 3775
chlorpropamide              3775
glimepiride                 3775
acetohexamide               3775
glipizide 

In [10]:
#filter by age group 20-30
df[df['age']=='[20-30)'].count()

encounter_id                1657
patient_nbr                 1657
race                        1657
gender                      1657
age                         1657
weight                      1657
admission_type_id           1657
discharge_disposition_id    1657
admission_source_id         1657
time_in_hospital            1657
payer_code                  1657
medical_specialty           1657
num_lab_procedures          1657
num_procedures              1657
num_medications             1657
number_outpatient           1657
number_emergency            1657
number_inpatient            1657
diag_1                      1657
diag_2                      1657
diag_3                      1657
number_diagnoses            1657
max_glu_serum               1657
A1Cresult                   1657
metformin                   1657
repaglinide                 1657
nateglinide                 1657
chlorpropamide              1657
glimepiride                 1657
acetohexamide               1657
glipizide 

In [11]:
#filter by age group 10-20
df[df['age']=='[10-20)'].count()

encounter_id                691
patient_nbr                 691
race                        691
gender                      691
age                         691
weight                      691
admission_type_id           691
discharge_disposition_id    691
admission_source_id         691
time_in_hospital            691
payer_code                  691
medical_specialty           691
num_lab_procedures          691
num_procedures              691
num_medications             691
number_outpatient           691
number_emergency            691
number_inpatient            691
diag_1                      691
diag_2                      691
diag_3                      691
number_diagnoses            691
max_glu_serum               691
A1Cresult                   691
metformin                   691
repaglinide                 691
nateglinide                 691
chlorpropamide              691
glimepiride                 691
acetohexamide               691
glipizide                   691
glyburid

In [12]:
#filter by age group 0-10
df[df['age']=='[0-10)'].count()

encounter_id                161
patient_nbr                 161
race                        161
gender                      161
age                         161
weight                      161
admission_type_id           161
discharge_disposition_id    161
admission_source_id         161
time_in_hospital            161
payer_code                  161
medical_specialty           161
num_lab_procedures          161
num_procedures              161
num_medications             161
number_outpatient           161
number_emergency            161
number_inpatient            161
diag_1                      161
diag_2                      161
diag_3                      161
number_diagnoses            161
max_glu_serum               161
A1Cresult                   161
metformin                   161
repaglinide                 161
nateglinide                 161
chlorpropamide              161
glimepiride                 161
acetohexamide               161
glipizide                   161
glyburid

In [14]:
#filter by age group 70-80
df[df['age']=='[70-80)'].count()

encounter_id                26068
patient_nbr                 26068
race                        26068
gender                      26068
age                         26068
weight                      26068
admission_type_id           26068
discharge_disposition_id    26068
admission_source_id         26068
time_in_hospital            26068
payer_code                  26068
medical_specialty           26068
num_lab_procedures          26068
num_procedures              26068
num_medications             26068
number_outpatient           26068
number_emergency            26068
number_inpatient            26068
diag_1                      26068
diag_2                      26068
diag_3                      26068
number_diagnoses            26068
max_glu_serum               26068
A1Cresult                   26068
metformin                   26068
repaglinide                 26068
nateglinide                 26068
chlorpropamide              26068
glimepiride                 26068
acetohexamide 

In [15]:
#filter by age group 80-90
df[df['age']=='[80-90)'].count()

encounter_id                17197
patient_nbr                 17197
race                        17197
gender                      17197
age                         17197
weight                      17197
admission_type_id           17197
discharge_disposition_id    17197
admission_source_id         17197
time_in_hospital            17197
payer_code                  17197
medical_specialty           17197
num_lab_procedures          17197
num_procedures              17197
num_medications             17197
number_outpatient           17197
number_emergency            17197
number_inpatient            17197
diag_1                      17197
diag_2                      17197
diag_3                      17197
number_diagnoses            17197
max_glu_serum               17197
A1Cresult                   17197
metformin                   17197
repaglinide                 17197
nateglinide                 17197
chlorpropamide              17197
glimepiride                 17197
acetohexamide 

In [16]:
#filter by age group 90-100
df[df['age']=='[90-100)'].count()

encounter_id                2793
patient_nbr                 2793
race                        2793
gender                      2793
age                         2793
weight                      2793
admission_type_id           2793
discharge_disposition_id    2793
admission_source_id         2793
time_in_hospital            2793
payer_code                  2793
medical_specialty           2793
num_lab_procedures          2793
num_procedures              2793
num_medications             2793
number_outpatient           2793
number_emergency            2793
number_inpatient            2793
diag_1                      2793
diag_2                      2793
diag_3                      2793
number_diagnoses            2793
max_glu_serum               2793
A1Cresult                   2793
metformin                   2793
repaglinide                 2793
nateglinide                 2793
chlorpropamide              2793
glimepiride                 2793
acetohexamide               2793
glipizide 

In [19]:
#filter by gender how many females are in the study
df[df['gender'] == 'Female'].count()

encounter_id                54708
patient_nbr                 54708
race                        54708
gender                      54708
age                         54708
weight                      54708
admission_type_id           54708
discharge_disposition_id    54708
admission_source_id         54708
time_in_hospital            54708
payer_code                  54708
medical_specialty           54708
num_lab_procedures          54708
num_procedures              54708
num_medications             54708
number_outpatient           54708
number_emergency            54708
number_inpatient            54708
diag_1                      54708
diag_2                      54708
diag_3                      54708
number_diagnoses            54708
max_glu_serum               54708
A1Cresult                   54708
metformin                   54708
repaglinide                 54708
nateglinide                 54708
chlorpropamide              54708
glimepiride                 54708
acetohexamide 

In [20]:
#filter by gender how many males are in the study
df[df['gender'] == 'Male'].count()

encounter_id                47055
patient_nbr                 47055
race                        47055
gender                      47055
age                         47055
weight                      47055
admission_type_id           47055
discharge_disposition_id    47055
admission_source_id         47055
time_in_hospital            47055
payer_code                  47055
medical_specialty           47055
num_lab_procedures          47055
num_procedures              47055
num_medications             47055
number_outpatient           47055
number_emergency            47055
number_inpatient            47055
diag_1                      47055
diag_2                      47055
diag_3                      47055
number_diagnoses            47055
max_glu_serum               47055
A1Cresult                   47055
metformin                   47055
repaglinide                 47055
nateglinide                 47055
chlorpropamide              47055
glimepiride                 47055
acetohexamide 

In [25]:
table=pd.pivot_table(
data=df,
index='age',
values='time_in_hospital',
aggfunc='mean')
table

time_in_hospital
age                       
[0-10)            2.546584
[10-20)           3.191027
[20-30)           3.564876
[30-40)           3.799735
[40-50)           4.039649
[50-60)           4.125753
[60-70)           4.382244
[70-80)           4.590878
[80-90)           4.808629
[90-100)          4.755818

In [ ]:
table.plot()

In [26]:
df.describe()

encounter_id   patient_nbr  admission_type_id  \
count  1.017660e+05  1.017660e+05      101766.000000   
mean   1.652016e+08  5.433040e+07           2.024006   
std    1.026403e+08  3.869636e+07           1.445403   
min    1.252200e+04  1.350000e+02           1.000000   
25%    8.496119e+07  2.341322e+07           1.000000   
50%    1.523890e+08  4.550514e+07           1.000000   
75%    2.302709e+08  8.754595e+07           3.000000   
max    4.438672e+08  1.895026e+08           8.000000   

       discharge_disposition_id  admission_source_id  time_in_hospital  \
count             101766.000000        101766.000000     101766.000000   
mean                   3.715642             5.754437          4.395987   
std                    5.280166             4.064081          2.985108   
min                    1.000000             1.000000          1.000000   
25%                    1.000000             1.000000          2.000000   
50%                    1.000000             7.000000          4.000000   
75%                    4.000000             7.000000          6.000000   
max                   28.000000            25.000000         14.000000   

       num_lab_procedures  num_procedures  num_medications  number_outpatient  \
count       101766.000000   101766.000000    101766.000000      101766.000000   
mean            43.095641        1.339730        16.021844           0.369357   
std             19.674362        1.705807         8.127566           1.267265   
min              1.000000        0.000000         1.000000           0.000000   
25%             31.000000        0.000000        10.000000           0.000000   
50%             44.000000        1.000000        15.000000           0.000000   
75%             57.000000        2.000000        20.000000           0.000000   
max            132.000000        6.000000        81.000000          42.000000   

       number_emergency  number_inpatient  number_diagnoses  
count     101766.000000     101766.000000     101766.000000  
mean           0.197836          0.635566          7.422607  
std            0.930472          1.262863          1.933600  
min            0.000000          0.000000          1.000000  
25%            0.000000          0.000000          6.000000  
50%            0.000000          0.000000          8.000000  
75%            0.000000          1.000000          9.000000  
max           76.000000         21.000000         16.000000

In [27]:
df.describe(include=['object'])

race  gender      age  weight payer_code medical_specialty  \
count      101766  101766   101766  101766     101766            101766   
unique          6       3       10      10         18                73   
top     Caucasian  Female  [70-80)       ?          ?                 ?   
freq        76099   54708    26068   98569      40256             49949   

        diag_1  diag_2  diag_3 max_glu_serum  ... citoglipton insulin  \
count   101766  101766  101766        101766  ...      101766  101766   
unique     717     749     790             4  ...           1       4   
top        428     276     250          None  ...          No      No   
freq      6862    6752   11555         96420  ...      101766   47383   

       glyburide-metformin glipizide-metformin glimepiride-pioglitazone  \
count               101766              101766                   101766   
unique                   4                   2                        2   
top                     No                  No                       No   
freq                101060              101753                   101765   

       metformin-rosiglitazone metformin-pioglitazone  change diabetesMed  \
count                   101766                 101766  101766      101766   
unique                       2                      2       2           2   
top                         No                     No      No         Yes   
freq                    101764                 101765   54755       78363   

       readmitted  
count      101766  
unique          3  
top            NO  
freq        54864  

[4 rows x 37 columns]

In [29]:
#show the type of dat I have in each column, how many data points there are, how much memry the data takes up
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101766 entries, 0 to 101765
Data columns (total 50 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   encounter_id              101766 non-null  int64 
 1   patient_nbr               101766 non-null  int64 
 2   race                      101766 non-null  object
 3   gender                    101766 non-null  object
 4   age                       101766 non-null  object
 5   weight                    101766 non-null  object
 6   admission_type_id         101766 non-null  int64 
 7   discharge_disposition_id  101766 non-null  int64 
 8   admission_source_id       101766 non-null  int64 
 9   time_in_hospital          101766 non-null  int64 
 10  payer_code                101766 non-null  object
 11  medical_specialty         101766 non-null  object
 12  num_lab_procedures        101766 non-null  int64 
 13  num_procedures            101766 non-null  int64 
 14  num_

In [31]:
# Check for null values in each column
null_counts = df.isnull().sum()
null_counts

encounter_id                0
patient_nbr                 0
race                        0
gender                      0
age                         0
weight                      0
admission_type_id           0
discharge_disposition_id    0
admission_source_id         0
time_in_hospital            0
payer_code                  0
medical_specialty           0
num_lab_procedures          0
num_procedures              0
num_medications             0
number_outpatient           0
number_emergency            0
number_inpatient            0
diag_1                      0
diag_2                      0
diag_3                      0
number_diagnoses            0
max_glu_serum               0
A1Cresult                   0
metformin                   0
repaglinide                 0
nateglinide                 0
chlorpropamide              0
glimepiride                 0
acetohexamide               0
glipizide                   0
glyburide                   0
tolbutamide                 0
pioglitazo

In [6]:
Diagnosis=df['diag_1'].count()
Diagnosis

101766

In [8]:
df.groupby(['diag_1'])['patient_nbr'].count().sort_values(ascending=False)

diag_1
428    6862
414    6581
786    4016
410    3614
486    3508
       ... 
832       1
833       1
919       1
834       1
10        1
Name: patient_nbr, Length: 717, dtype: int64

diagnosis 428: is congestive heart failure, its the most frequent diagnosis.


In [9]:
df.groupby(['diag_2'])['patient_nbr'].count().sort_values(ascending=False)

diag_2
276    6752
428    6662
250    6071
427    5036
401    3736
       ... 
388       1
994       1
800       1
46        1
811       1
Name: patient_nbr, Length: 749, dtype: int64

diagnosis 276 : hyposmolality and or hyponatremia

In [10]:
df.groupby(['diag_3'])['patient_nbr'].count().sort_values(ascending=False)

diag_3
250     11555
401      8289
276      5175
428      4577
427      3955
        ...  
972         1
E886        1
980         1
448         1
744         1
Name: patient_nbr, Length: 790, dtype: int64

In [ ]:
diagnosis 11555: is diabetes mellitus

In [11]:
list(df)

['encounter_id',
 'patient_nbr',
 'race',
 'gender',
 'age',
 'weight',
 'admission_type_id',
 'discharge_disposition_id',
 'admission_source_id',
 'time_in_hospital',
 'payer_code',
 'medical_specialty',
 'num_lab_procedures',
 'num_procedures',
 'num_medications',
 'number_outpatient',
 'number_emergency',
 'number_inpatient',
 'diag_1',
 'diag_2',
 'diag_3',
 'number_diagnoses',
 'max_glu_serum',
 'A1Cresult',
 'metformin',
 'repaglinide',
 'nateglinide',
 'chlorpropamide',
 'glimepiride',
 'acetohexamide',
 'glipizide',
 'glyburide',
 'tolbutamide',
 'pioglitazone',
 'rosiglitazone',
 'acarbose',
 'miglitol',
 'troglitazone',
 'tolazamide',
 'examide',
 'citoglipton',
 'insulin',
 'glyburide-metformin',
 'glipizide-metformin',
 'glimepiride-pioglitazone',
 'metformin-rosiglitazone',
 'metformin-pioglitazone',
 'change',
 'diabetesMed',
 'readmitted']

In [28]:
df2=df[['patient_nbr','diag_1', 'diag_2', 'diag_3', 'patient_nbr', 'race', 'gender', 'age', 'weight','number_emergency']]

In [29]:
df2

patient_nbr  diag_1  diag_2 diag_3  patient_nbr             race  \
0           8222157  250.83       ?      ?      8222157        Caucasian   
1          55629189     276  250.01    255     55629189        Caucasian   
2          86047875     648     250    V27     86047875  AfricanAmerican   
3          82442376       8  250.43    403     82442376        Caucasian   
4          42519267     197     157    250     42519267        Caucasian   
...             ...     ...     ...    ...          ...              ...   
101761    100162476  250.13     291    458    100162476  AfricanAmerican   
101762     74694222     560     276    787     74694222  AfricanAmerican   
101763     41088789      38     590    296     41088789        Caucasian   
101764     31693671     996     285    998     31693671        Caucasian   
101765    175429310     530     530    787    175429310        Caucasian   

        gender      age weight  number_emergency  
0       Female   [0-10)      ?                 0  
1       Female  [10-20)      ?                 0  
2       Female  [20-30)      ?                 0  
3         Male  [30-40)      ?                 0  
4         Male  [40-50)      ?                 0  
...        ...      ...    ...               ...  
101761    Male  [70-80)      ?                 0  
101762  Female  [80-90)      ?                 0  
101763    Male  [70-80)      ?                 0  
101764  Female  [80-90)      ?                 0  
101765    Male  [70-80)      ?                 0  

[101766 rows x 10 columns]

In [23]:
age=df2['age'].unique().tolist()

In [24]:
age

['[0-10)',
 '[10-20)',
 '[20-30)',
 '[30-40)',
 '[40-50)',
 '[50-60)',
 '[60-70)',
 '[70-80)',
 '[80-90)',
 '[90-100)']

In [25]:
weight=df2['weight'].unique().tolist()

In [26]:
weight

['?',
 '[75-100)',
 '[50-75)',
 '[0-25)',
 '[100-125)',
 '[25-50)',
 '[125-150)',
 '[175-200)',
 '[150-175)',
 '>200']

In [17]:
df.columns

Index(['encounter_id', 'patient_nbr', 'race', 'gender', 'age', 'weight',
       'admission_type_id', 'discharge_disposition_id', 'admission_source_id',
       'time_in_hospital', 'payer_code', 'medical_specialty',
       'num_lab_procedures', 'num_procedures', 'num_medications',
       'number_outpatient', 'number_emergency', 'number_inpatient', 'diag_1',
       'diag_2', 'diag_3', 'number_diagnoses', 'max_glu_serum', 'A1Cresult',
       'metformin', 'repaglinide', 'nateglinide', 'chlorpropamide',
       'glimepiride', 'acetohexamide', 'glipizide', 'glyburide', 'tolbutamide',
       'pioglitazone', 'rosiglitazone', 'acarbose', 'miglitol', 'troglitazone',
       'tolazamide', 'examide', 'citoglipton', 'insulin',
       'glyburide-metformin', 'glipizide-metformin',
       'glimepiride-pioglitazone', 'metformin-rosiglitazone',
       'metformin-pioglitazone', 'change', 'diabetesMed', 'readmitted'],
      dtype='object')

In [21]:
med_specialty=df['medical_specialty']
unique_med=med_specialty.unique()
unique_med

array(['Pediatrics-Endocrinology', '?', 'InternalMedicine',
       'Family/GeneralPractice', 'Cardiology', 'Surgery-General',
       'Orthopedics', 'Gastroenterology',
       'Surgery-Cardiovascular/Thoracic', 'Nephrology',
       'Orthopedics-Reconstructive', 'Psychiatry', 'Emergency/Trauma',
       'Pulmonology', 'Surgery-Neuro',
       'Obsterics&Gynecology-GynecologicOnco', 'ObstetricsandGynecology',
       'Pediatrics', 'Hematology/Oncology', 'Otolaryngology',
       'Surgery-Colon&Rectal', 'Pediatrics-CriticalCare', 'Endocrinology',
       'Urology', 'Psychiatry-Child/Adolescent', 'Pediatrics-Pulmonology',
       'Neurology', 'Anesthesiology-Pediatric', 'Radiology',
       'Pediatrics-Hematology-Oncology', 'Psychology', 'Podiatry',
       'Gynecology', 'Oncology', 'Pediatrics-Neurology',
       'Surgery-Plastic', 'Surgery-Thoracic',
       'Surgery-PlasticwithinHeadandNeck', 'Ophthalmology',
       'Surgery-Pediatric', 'Pediatrics-EmergencyMedicine',
       'PhysicalMedicineandRe

In [23]:
#what were the most visited medical specialty
df.groupby(['medical_specialty'])['encounter_id'].count().sort_values(ascending=False)

medical_specialty
?                         49949
InternalMedicine          14635
Emergency/Trauma           7565
Family/GeneralPractice     7440
Cardiology                 5352
                          ...  
Proctology                    1
Psychiatry-Addictive          1
Neurophysiology               1
SportsMedicine                1
Perinatology                  1
Name: encounter_id, Length: 73, dtype: int64

In [25]:
#list of payer_codes
payer_code=df['payer_code']
unique_payer=payer_code.unique()
unique_payer

array(['?', 'MC', 'MD', 'HM', 'UN', 'BC', 'SP', 'CP', 'SI', 'DM', 'CM',
       'CH', 'PO', 'WC', 'OT', 'OG', 'MP', 'FR'], dtype=object)

In [ ]:
#visualization 
# Group by 'payer_code' and count the unique 'encounter_id'
payer_counts = df.groupby('payer_code')['encounter_id'].count()

# Create the bar plot
plt.figure(figsize=(10, 6))
payer_counts.plot(kind='bar')
plt.xlabel('Payer Code')
plt.ylabel('Count of Encounter ID')
plt.title('Number of Encounters per Payer Code')
plt.xticks(rotation=0)
plt.show()

In [ ]:
plt.figure(figsize=(8, 6))
race_counts = df['race'].value_counts()
plt.bar(race_counts.index, race_counts.values)
plt.xlabel('Race')
plt.ylabel('Count')
plt.title('Race Distribution')
plt.show()

In [ ]:
plt.figure(figsize=(6, 6))
gender_counts = df['gender'].value_counts()
plt.pie(gender_counts.values, labels=gender_counts.index, autopct='%1.1f%%', startangle=90)
plt.title('Gender Distribution')
plt.axis('equal')  # Equal aspect ratio ensures the pie is drawn as a circle.
plt.show()

In [ ]:
plt.figure(figsize=(8, 6))
time_in_hospital_counts = df['time_in_hospital'].value_counts().sort_index()
plt.bar(time_in_hospital_counts.index, time_in_hospital_counts.values)
plt.xlabel('Time in Hospital')
plt.ylabel('Count')
plt.title('Time in Hospital Distribution')
plt.show()

The diabetic data set has  
101766 rows and 50 columns     
In the age group 90-100 there were 2793 individuals  
In the age group 80-90 there were 17197 individuals  
In the age group 70-80 there were 26068 individuals  
In the age group 60-70 there were 22483 individuals  
In the age group 50-60 there were 17256 individuals  
In the age group 40-50 there were 9685 individuals  
In the age group 30-40 there were 3775 individuals  
In the age group 20-30 there were 1657 individuals  
In the age group 10-20 there were 691 individuals  
In the age group 0-10 there were  161 individuals  

There were 54708 females and 47055 males  
The age group that spent most time in the hospital were   
1)	80-90 years  
2)	90-100 years  
3)	70-80 years  
The columns did not have null values  
#1 diagnosis patients came for was 428: congestive heart failure  
#2 diagnosis patients came for was 276: hyposmolality and or hyponatremia  
#3 diagnosis patients came for was 250: diabetes mellitus  

The number 1 medical specialty is unknown because it had a question mark,  
The number 2 medical specialty is internal medicine  
The number 1 medical specialty is emergency/trauma  
